### PR TITLE
Clean up deprecated features

### PR DESCRIPTION
**Pull request by nuka137**
## Motivation

See #2902

## Description of the changes

Remove deprecated features as listed below.

* `LightGBMTuner::best_booster`
* `FrozenTrial`
* `StudySummary`
* `TrialPruned`

### CODE REVIEW

**Comment by himkt** [comment]
Thank you for the PR! Please remove tests using them as well. :pray:

File: tests/integration_tests/lightgbm_tuner_tests/test_optimize.py [hunk 1]
```diff
@@ -640,10 +640,6 @@ def test_get_best_booster(self) -> None:
         best_booster = tuner.get_best_booster()
         assert best_booster.params["lambda_l1"] != unexpected_value
 
-        # TODO(toshihikoyanase): Remove this check when LightGBMTuner.best_booster is removed.
```
Comment (himkt) [inline tests/integration_tests/lightgbm_tuner_tests/test_optimize.py:643]: :+1:

File: tests/study_tests/test_optimize.py [hunk 2]
```diff
@@ -58,7 +57,7 @@ def func_nan(_: Trial) -> float:
 # TODO(Yanase): Remove this test function after removing `optuna.structs.TrialPruned`.
```
Comment (himkt) [inline tests/study_tests/test_optimize.py:57]: I'm wondering if we can remove this test as @toshihikoyanase said here, as well. cc. @toshihikoyanase

Comment (nuka137) [reply tests/study_tests/test_optimize.py]: Thanks! I deleted this test

**Comment by nuka137** [comment]
@himkt 

After your review, I deleted lazy import in `optuna/__init__.py` which is not needed any more.
Could you also check this?

**Review by himkt — APPROVED** [review]
LGTM!

**Review by not522 — APPROVED** [review]
LGTM!

**Pull request merged**